src/coreclr/jit/compiler.h

```diff
@@ -11490,42 +11491,14 @@ class GenTreeVisitor
                 break;
             }
 
-            case GT_DYN_BLK:
-            {
-                GenTreeDynBlk* const dynBlock = node->AsDynBlk();
-
-                GenTree** op1Use = &dynBlock->gtOp1;
-                GenTree** op2Use = &dynBlock->gtDynamicSize;
-
-                result = WalkTree(op1Use, dynBlock);
-                if (result == fgWalkResult::WALK_ABORT)
-                {
-                    return result;
-                }
-                result = WalkTree(op2Use, dynBlock);
-                if (result == fgWalkResult::WALK_ABORT)
-                {
-                    return result;
-                }
-                break;
-            }
-
             case GT_STORE_DYN_BLK:
             {
-                GenTreeDynBlk* const dynBlock = node->AsDynBlk();
+                GenTreeStoreDynBlk* const dynBlock = node->AsStoreDynBlk();
 
                 GenTree** op1Use = &dynBlock->gtOp1;
                 GenTree** op2Use = &dynBlock->gtOp2;
                 GenTree** op3Use = &dynBlock->gtDynamicSize;
 
-                if (TVisitor::UseExecutionOrder)
-                {
-                    if (dynBlock->IsReverseOp())
-                    {
-                        std::swap(op1Use, op2Use);
-                    }
-                }
-
```

Comment:
Maybe I haven't seen it yet, but why get rid of the reverse ops option? Isn't that used to improve codegen if the order doesn't matter, so we can execute the heavier op first?

Reply:
This was dead code - nothing sets `GTF_REVERSE_OPS` on a `STORE_DYN_BLK` node. `GTF_REVERSE_OPS` is only supported for (most) binary nodes and 2-operand `GenTreeMultiOp`s.

However, what did happen is that we reversed `ASG`s that would later turn into `STORE_DYN_BLK`s and now we don't. The handling could be added, but it would have to be "non-standard" because `STORE_DYN_BLK` is a tertiary operator, and I believe, based on the diffs for this change, that it would not be worth it at this point.

Side note: we could get all the 6 permutations of order "for free" if we changed `STORE_DYN_BLK` from a tertiary operator with `DynamicSize` as its member into two binary operators: `STORE_DYN_BLK(DstAddr, Data)` where `Data` would be `DYN_BLK(Addr, Size) | DYN_BLK_INIT_VAL(InitVal, Size)`, and also solve the problem with the "dummy" `IND struct` node. But as I note in the description it would be a larger change with quite a bit of churn to it.